### PR TITLE
NMS-14855: added a policy and a detector to Minion HeartBeatConsumer …

### PR DESCRIPTION
**Pull request by aramosvizcarra**
…to ignore docker interfaces and detect snmp agent

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14855

### CODE REVIEW

File: features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java [hunk 1]
```diff
@@ -216,16 +221,30 @@ private void provision(final OnmsMinion minion,
             return;
         }
 
+        final String prevForeignSource = String.format(PROVISIONING_FOREIGN_SOURCE_PATTERN, prevLocation);
+        final String nextForeignSource = String.format(PROVISIONING_FOREIGN_SOURCE_PATTERN, nextLocation);
+
+        PluginConfig policy = new PluginConfig(DEFAULT_SNMP_POLICY, "org.opennms.netmgt.provision.persist.policies.MatchingIpInterfacePolicy");
+        policy.addParameter("ipAddress", "~^docker.*$");
+        policy.addParameter("action", "DO_NOT_PERSIST");
```
Comment (christianpape) [inline features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java:229]: The IPAddress will never match docker.* here. I think you want to use the MatchingSnmpInterfacePolicy here, right?

**Review by christianpape — CHANGES_REQUESTED** [review]

File: features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java (hunk 1 above)
Comment (j-white) [reply features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java:229]: An example here: https://opennms.discourse.group/t/ignoring-docker-volumes-and-network-interfaces/1189

Comment (aramosvizcarra) [reply features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java]: > An example here: https://opennms.discourse.group/t/ignoring-docker-volumes-and-network-interfaces/1189

Thanks, bookmarked

Comment (aramosvizcarra) [reply features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java]: > The IP-Address will never match docker.* here. I think you want to use the MatchingSnmpInterfacePolicy here and match against ifDescr, right?

Thanks, I didn't know and couldn't find any documentation about it. Now I have some...

**Review by christianpape — APPROVED** [review]
LGTM.

**Review by fooker — APPROVED** [review]

**Comment by Bonrob2** [comment]
Not sure why the bot tagged me for review.

**Review by christianpape — APPROVED** [review]
LGTM.

**Pull request merged**